casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala

```diff
@@ -91,6 +91,19 @@ object ProtoUtil {
       }
     } yield block
 
+  def unsafeGetBlockMetadata[F[_]: Monad](
+      hash: BlockHash,
+      dag: BlockDagRepresentation[F]
+  ): F[BlockMetadata] =
+    for {
+      maybeBlockMetadata <- dag.lookup(hash)
+      blockMetadata = maybeBlockMetadata match {
+        case Some(b) => b
+        case None =>
+          throw new Exception(s"DAG storage is missing hash ${PrettyPrinter.buildString(hash)}")
```

Comment:
throw is a no-go. Please replace it with `.raise`. I'm fine if you use a `:Sync` constraint in here - still a better love story than throws. Ideally it'd be `FunctorRaise[CasperError` though